component/background/RoomService.tsx

```diff
@@ -50,19 +54,17 @@ class RoomService {
     });
 
     // set up configRef
-    this.userListRef = this.rootRef.child("user");
-    this.myConfRef = this.userListRef.child(this.id);
-    this.myConfRef.set(true);
-
+    this.rootRef.child(USER).child(this.id).set(true);
+    this.myself = allUsers.child(this.id);
     this.active = true;
   }
 
   public close(): void {
     if (this.active) {
       this.messageRef.off();
-      this.myConfRef.remove()
+      this.rootRef.child(USER).child(this.id).remove()
         .then(() => {
-          this.userListRef.once("value", (data) => {
+          this.rootRef.child(USER).once("value", (data) => {
```

Comment:
I actually removed it to reduce the number of fields.
I already extract a UseService out of the Roomservice, because this part of the logic seems to serve a different purpose